lib/manager/gradle/index.ts

```diff
@@ -1,173 +1,29 @@
-    gradleWrapperFileName(config),
-    cwd,
-    gradlew,
-    GRADLE_DEPENDENCY_REPORT_OPTIONS
-  );
-  const execOptions: ExecOptions = {
-    timeout,
-    cwd,
-    docker: {
-      image: 'gradle',
-    },
-    extraEnv,
-  };
-  try {
-    logger.debug({ cmd }, 'Start gradle command');
-    ({ stdout, stderr } = await exec(cmd, execOptions));
-  } catch (err) /* istanbul ignore next */ {
-    if (err.message === TEMPORARY_ERROR) {
-      throw err;
-    }
-    if (err.code === TIMEOUT_CODE) {
-      throw new ExternalHostError(err, 'gradle');
-    }
-    logger.warn({ errMessage: err.message }, 'Gradle extraction failed');
-    return;
-  }
-  logger.debug(stdout + stderr);
-  logger.debug('Gradle report complete');
-}
-
-export async function extractAllPackageFiles(
+export function extractAllPackageFiles(
   config: ExtractConfig,
   packageFiles: string[]
 ): Promise<PackageFile[] | null> {
-  let rootBuildGradle: string | undefined;
-  let gradlew: Stats | null;
-  const { localDir } = getAdminConfig();
-  for (const packageFile of packageFiles) {
-    const dirname = upath.dirname(packageFile);
-    const gradlewPath = upath.join(dirname, gradleWrapperFileName(config));
-    gradlew = await stat(upath.join(localDir, gradlewPath)).catch(() => null);
-
-    if (['build.gradle', 'build.gradle.kts'].includes(packageFile)) {
-      rootBuildGradle = packageFile;
-      break;
-    }
-
-    // If there is gradlew in the same directory, the directory should be a Gradle project root
-    if (gradlew?.isFile() === true) {
-      rootBuildGradle = packageFile;
-      break;
-    }
-  }
-  if (!rootBuildGradle) {
-    logger.warn('No root build.gradle nor build.gradle.kts found - skipping');
-    return null;
-  }
-  logger.debug('Extracting dependencies from all gradle files');
-
-  const cwd = upath.join(localDir, upath.dirname(rootBuildGradle));
-
-  await createRenovateGradlePlugin(cwd);
-  await executeGradle(config, cwd, gradlew);
-
-  init();
-
-  const dependencies = await extractDependenciesFromUpdatesReport(cwd);
-  if (dependencies.length === 0) {
-    return [];
-  }
-
-  const gradleFiles: PackageFile[] = [];
-  for (const packageFile of packageFiles) {
-    const content = await readLocalFile(packageFile, 'utf8');
-    if (content) {
-      gradleFiles.push({
-        packageFile,
-        datasource: datasourceMaven.id,
-        deps: dependencies,
-      });
-
-      collectVersionVariables(dependencies, content);
-    } else {
-      // istanbul ignore next
-      logger.debug({ packageFile }, 'packageFile has no content');
-    }
-  }
-
-  return gradleFiles;
-}
-
-function buildGradleDependency(config: Upgrade): GradleDependency {
-  return {
-    group: config.depGroup,
-    name: config.name,
-    version: config.currentValue,
-  };
+  return config.deepExtract
+    ? deep.extractAllPackageFiles(config, packageFiles)
+    : shallow.extractAllPackageFiles(config, packageFiles);
```

Comment:
That's good it's only can be either deep or shallow, not both. I doubt the reasonable mixed solution is possible.